libbz2-rs-sys/src/decompress.rs

```diff
@@ -1009,9 +1009,40 @@ pub(crate) fn decompress(
                                 s.selector[i] = pos[0];
                             }
 
-                            t = 0;
-                            current_block = Block35;
-                            break;
+                            // try to read the coding tables in one go if there is sufficient input
+                            let bits_needed =
+                                usize::from(nGroups) * (5 + (usize::from(alphaSize) * 2));
+                            let bytes_needed = bits_needed.div_ceil(8);
+
+                            if strm.avail_in as usize >= bytes_needed {
+                                for t in 0..nGroups {
+                                    let mut curr = GET_BITS!(strm, s, 5);
+                                    for i in 0..alphaSize {
+                                        loop {
```

Comment:
it should be possible to bound this loop (I think it's 20 bits of code at most, so 40bits total?). But it's hard to be confident